### ISSUE TITLE
blas/cblas performance impact: Apple Accelerate can speedup on arm64 

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#5040]
## 🐛 Bug

@psobolewskiPhD observed a slowdown in point selected/drag behaviour in #5022 which appears to be tied to a specific environment rather than any specific napari commit - it would be good to understand the source of this

**Comment by alisterburt** [comment]
https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Napari.20.26.20Python.20on.20Apple.20M1/near/297724448
```quote
In case anyone is curious, I noticed, what I thought was a major performance regression in my dev env (using main and various PR). Pan-zoom, moving points, whatever was choppier than normal and CPU usage 200-400%, vs my normal live env of butter smooth and ~40% CPU. Comparing the envs, I noticed that the performant env has:

libblas                   3.9.0           16_osxarm64_accelerate    conda-forge
libcblas                  3.9.0           16_osxarm64_accelerate    conda-forge
and the non-performant has:

libblas                   3.9.0           15_osxarm64_openblas    conda-forge
libcblas                  3.9.0           15_osxarm64_openblas    conda-forge
Making a fresh env with accelerate blas ( mamba install numpy "libblas=*=*accelerate" ) and then napari vs. a "normal" fresh env with napari makes the issue clear.
Doing mamba install numpy "libblas=*=*accelerate" into my dev env also made things smooth.
Pretty crazy.
```

**Comment by psobolewskiPhD** [comment]
I think based on the analysis, it's fair to say it's not a slowdown, but speed up?
I think the "normal" numpy should be the baseline. 
But the Accelerate framework on arm64 is markedly more performant.

**Comment by alisterburt** [comment]
you're right! we should add a note to the installation docs for m1 users so they can be aware of this :)

**Comment by psobolewskiPhD** [comment]
Before modifying anything, I'm curious how the situation is for x86 macOS users with a default conda-forge install. I'd consider using Accelerate BLAS experimental at this point. I don't think, for example, that scipy supports it.
Re: M1 installation, I think it may be worth adding a line somewhere, that arm64 native support requires pyqt/pyside from conda-forge (once Qt6 is not-experimental then pip is fine).

**Comment by Czaki** [comment]
Looks like a really apple thing. Instead of improve standard we will release propertiary solution that could be used only on our system.

**Comment by psobolewskiPhD** [comment]
on M1 with macOS 14 LAPACK is also accelerated.
Can install via pip now too:
`pip install -U --no-cache-dir --no-binary :all: numpy`
This will be default soon?
https://github.com/numpy/numpy/issues/24905

I see a pretty major drop in CPU usage when e.g. spastically drawing labels or spastically moving groups of points.
(note at present this is not the same as the conda-forge option, which only has Accelerate BLAS)

**Comment by psobolewskiPhD** [comment]
With numpy 2.0 on pypi, this requires nothing special now.
conda-forge is still a work in progress.

**Comment by psobolewskiPhD** [comment]
With https://github.com/conda-forge/blas-feedstock/pull/105 merged, `libblas=*=*newaccelerate` can be used to access the improved BLAS/LAPACK on Apple Silicon. Note in my hands it results in massive reduction in CPU%, but not major gains in performance (10-15%?)
